src/neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -238,7 +238,7 @@ protected internal int GetInvocationCounter()
         /// <summary>
         /// The implementation of System.Runtime.GetRandom.
         /// Gets the random number genrated form the VRF
-        /// Primary geterates this random number with `prevHash`, the hash of the prevuous block
+        /// Primary geterates this random number with `prevHash`, the hash of the previous (validators.Length/3 + 1) block
```

Comment:
What's the difference?

Reply:
> What's the difference?

As you mentioned before, if we use the previous Block hash value as the feed, the primary might be able to control the value if the current consensus occurs view change, therefore, we should use the hash value of a block that is far away from the current index to make sure the dishonest primary could not know what the future Blockchain status is gonna be when it generates that hash value. https://github.com/neo-project/neo-modules/blob/f14c1a9d214bbdff43bb37ab33a09d76fc8c16d5/src/DBFTPlugin/ConsensusContext.cs#L342

Reply:
But the previous (`validators.Length/3 + 1`) block also has the opportunity to have the problem.

Reply:
We should look backwards for the nearest block with a different `primaryIndex`.

Reply:
> We should look backwards for the nearest block with a different `primaryIndex`.

Now I use the hash of the previous 1000th Block. In this way, no primary could know what the future status would be when it generates Block.